### ISSUE TITLE
System quazip 1.0 not detected during build

### ISSUE THREAD

**Issue by lgbaldoni** [keepassxreboot/keepassxc#5534]
## Overview
As per subject, cmake can't detect quazip 1.0

## Steps to Reproduce
1. Install quazip 1.0
2. Run cmake
3. cmake fails detecting system-installed quazip

## Expected Behavior
Quazip being detected during cmake configuration phase

## Actual Behavior
See log snippet:
```
  The package name passed to `find_package_handle_standard_args` (QUAZIP)
  does not match the name of the calling package (QuaZip).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/FindQuaZip.cmake:24 (find_package_handle_standard_args)
  src/keeshare/CMakeLists.txt:25 (find_package)
 This warning is for project developers.  Use -Wno-dev to suppress it.
-- Could NOT find QUAZIP (missing: QUAZIP_LIBRARIES QUAZIP_INCLUDE_DIR) 
 -- KeeShare: Secure container support is DISABLED; quazip library not found
```

## Context
Quazip now uses -Qt(5!6) as suffix instead of "5"

KeePassXC  2.6.1
Revision: Unknown

Operating System: linux openSUSE Tumbleweed
Desktop Env: KDE
Windowing System: X11

**Comment by droidmonkey** [comment]
Very poor decision by Quazip to change the package namr

**Comment by droidmonkey** [comment]
1.0 quazip is meant to be _co-installed_ with the 0.x version. We will not support 1.0 until it has robust support across distros. They are not ABI compatible.

**Comment by lgbaldoni** [comment]
@droidmonkey upstream quazip clarified the point in the 1.1 release. Will force keepassxc to use the 0.9 package.

**Comment by maresmar** [comment]
openSUSE has already quazip 1.1 so [KeePassXC only supports unsigned sharing](https://bugzilla.suse.com/show_bug.cgi?id=1181898) there... Do you plan to move to a new version or should I try to persuade them to builder older quazip as well :smile:

**Comment by aswild** [comment]
@maresmar [Arch Linux patches keepassxc](https://github.com/archlinux/svntogit-community/blob/packages/keepassxc/trunk/keepassxc-quazip1.patch) to compile against quazip 1.0 and 1.1, package maintainers for openSUSE and similar distros could pick up that patch if they wanted.

There's no source code changes needed, just the build system tweaks to find the new library name. I'm not sure what ABI compatibility concerns there are, since on Linux distros everything here is compiled from source.

**Comment by lgbaldoni** [comment]
> I'm not sure what ABI compatibility concerns there are, since on Linux distros everything here is compiled from source.

There might be an API incompatibility for all we know. And I would avoid taking requirements lightly when it comes to cryptography. So, it would be nice if @droidmonkey confirmed there are no security concerns in this.

**Comment by droidmonkey** [comment]
It's on my shortlist to get this incorporated. I also chatted with the quazip maintainers and they do not suspect a problem. The jump to 1.0 was more in name than in core changes. Personally I want to eliminate the dependency on quazip, and may end up doing that for 2.7.

**Comment by aswild** [comment]
> There might be an API incompatibility for all we know. And I would avoid taking requirements lightly when it comes to cryptography. 

That's a valid point, diligence is good when it comes to crypto code. I haven't looked much further than "yay it compiles, ship it" for my personal builds.

**Comment by michaelk83** [comment]
Fixed in #6560?
